lightning/src/util/events.rs

```diff
@@ -338,7 +338,20 @@ pub enum Event {
 		channel_id: [u8; 32],
 		/// The full transaction received from the user
 		transaction: Transaction
-	}
+	},
+	/// Indicates that a path for an outbound payment was successful.
+	///
+	/// Useful for scoring channels. See [`Event::PaymentSent`] for obtaining the payment preimage.
+	PaymentPathSuccessful {
+		/// The id returned by [`ChannelManager::send_payment`] and used with
+		/// [`ChannelManager::retry_payment`].
+		///
+		/// [`ChannelManager::send_payment`]: crate::ln::channelmanager::ChannelManager::send_payment
+		/// [`ChannelManager::retry_payment`]: crate::ln::channelmanager::ChannelManager::retry_payment
+		payment_id: PaymentId,
```

Comment:
We should probably add the `payment_hash` here, no?

Reply:
Yeah, I think so. Though when producing an event in `finalize_claims` we don't have the payment hash. I guess we could pull it from `PendingOutboundPayment` but we wouldn't have it for `Legacy` or `Fulfilled`. Or we could include it in `MonitorRestoreUpdates::finalized_claimed_htlcs`, but I think that would require changing `Channel::monitor_pending_finalized_fulfills`.

Seems we'd have to make it optional in `PaymentPathSuccessful` and change the serialization format for either `Channel` or `PendingOutboundPayment`. Let me know if I'm missing something, though.

Reply:
> I guess we could pull it from PendingOutboundPayment but we wouldn't have it for Legacy or Fulfilled.

That seems fine IMO. We'll want to drop support for those legacy versions probably sooner rather than later, just given the amount of overhead they create.